File: Kattis/Provinces and Gold.py
import sys
import math

def main():
    
    for i in sys.stdin:
        godlist = i.strip().split()

    gold = int(godlist[0])
    silver = int(godlist[1])
    copper = int(godlist[2])
    buyingpower = gold*3 + silver*2 + copper*1

    output = ""

    if buyingpower >= 2 and buyingpower < 5:
        output += "Estate"
    elif buyingpower >= 5 and buyingpower < 8:
        output += "Duchy"
    elif buyingpower >= 8:
        output += "Province"
    else:
        output = ""

    if output != "":
        output += " or "
    else:
        output = ""

    if buyingpower >= 6:
        output += "Gold"
    elif buyingpower < 6 and buyingpower >= 3:
        output += "Silver"
    else:
        output += "Copper"
    
    print(output)

main()

File: Kattis/provincesandgold.cpp
#include <iostream>
#include <string>
using namespace std;

int main(){

  int gold, silver, copper, buyingpower;
  string bestvictory;
  cin >> gold >> silver >> copper;

  buyingpower = gold*3 + silver*2 + copper*1;

  if (buyingpower >= 8)
    bestvictory.append("Province or ");
  else if (buyingpower < 8 && buyingpower >= 5)
    bestvictory.append("Duchy or ");
  else if (buyingpower < 5 && buyingpower >= 2)
    bestvictory.append("Estate or ");

  if (buyingpower >= 6)
    bestvictory.append("Gold");
  else if (buyingpower >= 3 && buyingpower < 6)
    bestvictory.append("Silver");
  else
    bestvictory.append("Copper");

  cout << bestvictory;

}